### PR TITLE
fix: Correct endDate inclusion by adding 1-day offset

### PR DESCRIPTION
**Pull request by shfc**
I discovered that when the parameter `until` is set to `endDate`, the event is not generated on that specific date.  So I made a 1-day adjustment to the date, and the event is now being generated as expected on both Firefox and Chrome.

### CODE REVIEW

**Comment by shfc** [comment]
I've investigated this bug further, and the cause is now clear. The issue arises when the course's start time exceeds the until time. For example, in the scenario shown below, the until time calculated by the original code doesn't cover the course's start time, so the event is not generated as expected.
image
After applying the fix, the until time is set to midnight after the endDate, ensuring that the last event occurrence is included, as shown in the image below.
image

Test Data:
```json
{
"B.SUBJECT":"COMP SCI",
"B.CATALOG_NBR":"7314",
"B.DESCR":"",
"D.XLATSHORTNAME":"Workshop",
"START_TIME":"11:00 AM",
"END_TIME":"12:00 PM",
"DATES":"04 Oct",
"G.DESCR":"Hughes",
"F.ROOM":"1111",
"F.DESCR":"Teaching Room",
"E.START_DT":"2024-10-04",
"E.END_DT":"2024-10-04",
}
```

File: src/parser/parser.ts [hunk 1]
```diff
@@ -43,11 +43,12 @@ function createCalEvent(calendar: ICalCalendar, data: { [x: string]: any }) {
...
   // Use this to calculate repeating times
-  let endDate = new Date(Date.parse(endDateData));
-  appendLog(logLevels.DEBUG,`Creating event: ${eventSummary} - start: ${eventStart}; end: ${eventEnd}; enddate: ${endDate}; clsType: ${classType}`);
+  // The UNTIL parameter is non-inclusive, so set untilTime to midnight after the endDate to include the last event occurrence
+  let utilTime = new Date(new Date(endDateData).setHours(24, 0, 0, 0)); 
+  appendLog(logLevels.DEBUG,`Creating event: ${eventSummary} - start: ${eventStart}; end: ${eventEnd}; untiltime: ${utilTime}; clsType: ${classType}`);
   const repeatOptions: ICalRepeatingOptions = {
     freq: ICalEventRepeatingFreq.WEEKLY,
-    until: endDate,
+    until: utilTime,
```
Comment (rayokamoto) [inline src/parser/parser.ts:51]: Should be `untilTime` rather than `utilTime`?

**Review by rayokamoto — CHANGES_REQUESTED** [review]
Thanks for the test data, the issue is much clearer now. Missed this PR until the update just then, sorry for not reviewing earlier!

**Comment by shfc** [comment]
> Thanks for the test data, the issue is much clearer now. Missed this PR until the update just then, sorry for not reviewing earlier!

No worries! The original PR didn't fully address the root cause and still had potential issues. It's clearer now and easier to review.

**Review by rayokamoto — APPROVED** [review]

**Pull request merged**